.github/workflows/lint.yaml

```diff
@@ -4,6 +4,23 @@ on:
       - master
 
 jobs:
+  documentation:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # - https://bugs.launchpad.net/ubuntu/+source/mawk/+bug/69724
+      - run: awk -W version && awk '/[[:space:]]/ { matched=1 } END { exit !matched }' <<< ' '
```

Comment:
It always prints the awk version then exits non-zero when it does not match space to the posix class `[:space:]`.

The annoying part of trying to make this obvious is that awk "true" is non-zero while shell success is zero.
This needs an actual comment, so I'll add that. I think I'll also change the check to

```bash
awk '{ exit 1 != match($0, /[[:space:]]/) }' <<< ' '
```

The check is here at all because I was first trying this on Debian buster where `awk` is `mawk` and led me to the above Ubuntu issue. Of course, macOS has a different version/implementation and CentOS a different one again. 😖

Reply:
I was curious... These are all different from the GitHub action.

```console
$ docker run --rm -it debian:buster bash -c 'awk -W version'
mawk 1.3.3 Nov 1996, Copyright (C) Michael D. Brennan

compiled limits:
max NF             32767
sprintf buffer      2040

$ docker run --rm -it debian:latest bash -c 'awk -W version'
mawk 1.3.4 20200120
Copyright 2008-2019,2020, Thomas E. Dickey
Copyright 1991-1996,2014, Michael D. Brennan

random-funcs:       srandom/random
regex-funcs:        internal
compiled limits:
sprintf buffer      8192
maximum-integer     2147483647

$ docker run --rm ubuntu:latest bash -c 'awk -W version'
mawk 1.3.4 20200120
Copyright 2008-2019,2020, Thomas E. Dickey
Copyright 1991-1996,2014, Michael D. Brennan

random-funcs:       srandom/random
regex-funcs:        internal
compiled limits:
sprintf buffer      8192
maximum-integer     2147483647

$ docker run --rm centos:latest bash -c 'awk -W version'
GNU Awk 4.2.1, API: 2.0 (GNU MPFR 3.1.6-p2, GNU MP 6.1.2)
Copyright (C) 1989, 1991-2018 Free Software Foundation.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see http://www.gnu.org/licenses/.
```